library/std/src/sys/sync/rwlock/queue.rs

```diff
@@ -446,21 +471,21 @@ impl RwLock {
         // The state was observed with acquire ordering above, so the current
         // thread will observe all node initializations.
 
-        // SAFETY:
-        // Because new read-locks cannot be acquired while threads are queued,
-        // all queue-lock owners will observe the set `LOCKED` bit. Because no
-        // downgrade can be in progress (we checked above), they hence do not
-        // modify the queue, so the queue will not be removed from here.
+        // FIXME this is a bit confusing
+        // SAFETY: Because new read-locks cannot be acquired while threads are queued, all
+        // queue-lock owners will observe the set `LOCKED` bit. And because no downgrade can be in
+        // progress (we checked above), they hence do not modify the queue, so the queue will not be
+        // removed from here.
```

Comment:
> Because new read-locks cannot be acquired while threads are queued, all queue-lock owners will observe the set `LOCKED` bit.

This logic is a bit confusing, it's not super clear who the queue-lock owners are or why observing the `LOCKED` bit will prevent them from modifying the queue?